MarketOrganizer.Api/Interfaces/IAuthService.cs

```diff
@@ -0,0 +1,13 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using MarketOrganizer.Data.Models;
+
+namespace MarketOrganizer.Api.Interfaces
+{
+  public interface IAuthService<T>
+  {
+    Task<bool> Register(T user);
+    Task<bool> UserExists(string username);
+    User Login(string username, string password);
```

Comment:
podrias considerar dejar que toda la interfaz sea uniforme

```suggestion
    Task<User> Login(string username, string password);
```
despues de todo todos demas metodos en tu interfaz ya son async

Reply:
te refieres a que todos regresan un Task?

Reply:
sip, todos regresan un task